Makefile

```diff
@@ -118,11 +157,63 @@ build-musl-aux:
 		-DNGINX_COVERAGE=$(COVERAGE) \
 		&& cmake --build .musl-build -j $(MAKE_JOB_COUNT) -v
 
+NGINX_VERSION ?= $(shell echo ${RESTY_VERSION} | cut -d '.' -f 1-3)
+
+.PHONY: build-openresty-toolchain
+build-openresty-toolchain:
+ifndef RESTY_VERSION
+	$(error RESTY_VERSION is not set. Please set RESTY_VERSION environment variable)
+endif 
+ifndef RESTY_OPENSSL_VERSION
+	$(error RESTY_OPENSSL_VERSION is not set. Please set RESTY_OPENSSL_VERSION environment variable)
+endif 
+	docker build \
+		--build-arg RESTY_VERSION=${RESTY_VERSION} \
+		--build-arg RESTY_CONFIG_OPTIONS=${RESTY_CONFIG_OPTIONS} \
+		--build-arg RESTY_CONFIG_OPTIONS_MORE=${RESTY_CONFIG_OPTIONS_MORE} \
+		--build-arg RESTY_OPENSSL_URL_BASE=${RESTY_OPENSSL_URL_BASE} \
+		--build-arg RESTY_OPENSSL_VERSION=${RESTY_OPENSSL_VERSION} \
+		--build-arg RESTY_OPENSSL_PATCH_VERSION=${RESTY_OPENSSL_PATCH_VERSION} \
+		--build-arg RESTY_PCRE_VERSION=${RESTY_PCRE_VERSION} \
+		--build-arg RESTY_PCRE_BUILD_OPTIONS=${RESTY_PCRE_BUILD_OPTIONS} \
+		--build-arg RESTY_PCRE_SHA256=${RESTY_PCRE_SHA256} \
+		--build-arg RESTY_J=${RESTY_J} \
+		-t louis/openresty-${RESTY_VERSION} \
```

Comment:
If you have been unable to find a way for use to publicly publish images, WDYT about building the toolchain as part of the CI and therefore doesn't need to be published?

Reply:
I can publicly push images (I think at least), it is just that Damien wanted to give a look before publishing it. Building the toolchain in the CI would increase too much the duration of the CI.